### PR TITLE
Handle error in byte stream (failed transfer of a chunk)

### PR DESCRIPTION
**Pull request by pkopac**
**tldr; fixing large files transfer error is not reported**

My file transfers end successfully with partially transferred files.
To distinguish HTTP errors from transfer layer errors I've tested for `res == nil`, because (my logic went) such error won't return `Response`.
But then I investigated further and found [this explicitly ignored error](https://github.com/parnurzeal/gorequest/blob/develop/gorequest.go#L1121) (red flag). The `Body` is a stream that - for longer data - is chunked into multiple packets. Any of these can fail to transfer. Resuming aside, the program must know when this error happened.

### CODE REVIEW

**Comment by pkopac** [comment]
Seems like there's a problem with your configuration on Travis:
`An error occurred while generating the build script.`
The PR tested & build OK locally.
